build-info-extractor-maven3/src/main/java/org/jfrog/build/extractor/maven/BuildInfoRecorder.java

```diff
@@ -729,7 +730,16 @@ private void cleanUpModule() {
     /**
      * Skip the default maven deploy behaviour.
      */
-    private void skipDefaultDeploy(MavenSession session) {
-        session.getUserProperties().put("maven.deploy.skip", Boolean.TRUE.toString());
+    private void skipDeploy(ExecutionEvent event) {
+        List<String> goals = event.getSession().getRequest().getGoals();
+        // Override the default Maven deploy behavior with Artifactory.
+        if (goals.contains("deploy")) {
+            event.getSession().getUserProperties().put("maven.deploy.skip", Boolean.TRUE.toString());
+            return;
+        }
+        // Skip the artifact deployment behavior if the goals do not contain install phases.
+        if (!goals.contains("install")) {
+            this.conf.publisher.setBooleanValue(ADD_DEPLOYABLE_ARTIFACTS, false);
```

Comment:
We may still want the deployable artifacts list to allow late deployment
```suggestion
            conf.publisher.setPublishArtifacts(false);
```